.github/workflows/preview_build.yml

```diff
@@ -0,0 +1,132 @@
+    steps:
+    - name: Update deployment status to in progress
+      uses: octokit/request-action@v2.x
+      with:
+        route: POST /repos/{owner}/{repo}/deployments/{deployment_id}/statuses
+        mediaType: |
+          previews:
+          - ant-man
+          - flash
+        owner: ${{ github.event.repository.owner.login }}
+        repo: ${{ github.event.repository.name }}
+        deployment_id: ${{ github.event.client_payload.deployment_id }}
+        state: in_progress
+      env:
+        GITHUB_TOKEN: ${{ secrets.PREVIEW_GITHUB_TOKEN }}
+
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.client_payload.sha }}
+
+    - name: Name branch
+      run: |
+        name=$(echo "${{ github.event.client_payload.folder }}")
+        git checkout -b ${name}
+
+    - name: Setup cache
+      uses: actions/cache@v2
+      with:
+        path: /emsdk/upstream/emscripten/cache
+        key: 2.0.10-${{ runner.os }}
+
+    - name: Build (host tools)
+      run: |
+        mkdir build-host
+        cd build-host
+
+        echo "::group::CMake"
+        cmake .. -DOPTION_TOOLS_ONLY=ON
+        echo "::endgroup::"
+
+        echo "::group::Build"
+        echo "Running on $(nproc) cores"
+        make -j$(nproc) tools
+        echo "::endgroup::"
+
+    - name: Install GCC problem matcher
+      uses: ammaraskar/gcc-problem-matcher@master
+
+    - name: Build
+      run: |
+        mkdir build
+        cd build
+
+        echo "::group::CMake"
+        emcmake cmake .. \
+          -DHOST_BINARY_DIR=../build-host \
+          -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+          -DOPTION_USE_ASSERTS=OFF \
+          # EOF
+        echo "::endgroup::"
+
+        echo "::group::Build"
+        echo "Running on $(nproc) cores"
+        emmake make -j$(nproc)
+        echo "::endgroup::"
+
+    - name: Publish preview
+      run: |
+        pip3 install setuptools
+        pip3 install awscli
+
+        ~/.local/bin/aws s3 cp --only-show-errors build/openttd.data s3://${{ secrets.PREVIEW_S3_BUCKET }}/${{ github.event.client_payload.folder }}/
+        ~/.local/bin/aws s3 cp --only-show-errors build/openttd.html s3://${{ secrets.PREVIEW_S3_BUCKET }}/${{ github.event.client_payload.folder }}/
+        ~/.local/bin/aws s3 cp --only-show-errors build/openttd.js s3://${{ secrets.PREVIEW_S3_BUCKET }}/${{ github.event.client_payload.folder }}/
+        ~/.local/bin/aws s3 cp --only-show-errors build/openttd.wasm s3://${{ secrets.PREVIEW_S3_BUCKET }}/${{ github.event.client_payload.folder }}/
+
+        # Invalidate the cache of the CloudFront distribution
+        ~/.local/bin/aws cloudfront create-invalidation --distribution-id ${{ secrets.PREVIEW_CF_DISTRIBUTION_ID }} --paths "/${{ github.event.client_payload.folder }}/*"
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+    - name: Update deployment status to success
```

Comment:
I don't see why this manual "setting deployment status" is necessary - shouldn't it be the default based on the previous action. Can you explain?

Reply:
How deployments on GitHub work:

- You create a deployment -> this creates a "deployment" event
- You start a deployment (set status to pending)
- When it is done, you set the status to either success or failure

This step sets it to success, as we are done now, and the URL is available.

Does that answer the question a bit? :)